src/xenia/kernel/xboxkrnl/xboxkrnl_threading.cc

```diff
@@ -222,12 +222,15 @@ void KeSetCurrentStackPointers(lpvoid_t stack_ptr,
 }
 DECLARE_XBOXKRNL_EXPORT1(KeSetCurrentStackPointers, kThreading, kImplemented);
 
-dword_result_t KeSetAffinityThread(lpvoid_t thread_ptr, dword_t affinity) {
+dword_result_t KeSetAffinityThread(lpvoid_t thread_ptr, dword_t affinity,
+                                   lpdword_t previous_affinity_ptr) {
   auto thread = XObject::GetNativeObject<XThread>(kernel_state(), thread_ptr);
   if (thread) {
+    if (previous_affinity_ptr) {
+      *previous_affinity_ptr = 1 << thread->active_cpu();
+    }
     thread->SetAffinity(affinity);
   }
-
   return (uint32_t)affinity;
```

Comment:
The return value should be HRESULT according to the disassembly findings, please add handling of all the result codes from the original function that aren't too complex to integrate into the emulator's architecture (and if you skip any, leave some // comments about them for the possible implementation in the future).

Reply:
Also it would be good to have comments explaining the difference between the interfaces of KeSetAffinityThread on the Xbox 360 and on Windows, why there's a new parameter that isn't there in NT documentation or Wine, referencing the disassembly as the source.